### PR TITLE
feat: load the current database from Spring's application yaml

### PR DESCRIPTION
**Pull request by kmruiz**
### Checklist

- [ ] New tests and/or benchmarks are included.
- [ ] Documentation is changed or added.
- [ ] Changelog is updated accordingly.
- [ ] I have signed the MongoDB Contributor License Agreement (https://www.mongodb.com/legal/contributor-agreement).

### CODE REVIEW

File: packages/mongodb-dialects/spring-criteria/src/main/kotlin/com/mongodb/jbplugin/dialects/springcriteria/SpringCriteriaContextExtractor.kt [hunk 1]
```diff
@@ -14,13 +16,36 @@ object SpringCriteriaContextExtractor : ConnectionContextExtractor<Project> {
     )
 
     override fun gatherContext(contentRoot: Project): ConnectionContext {
-        val database = extractDatabase(contentRoot)
+        val database = extractDatabaseFromProperties(contentRoot)
+            ?: extractDatabaseFromYaml(contentRoot)
```
Comment (himanshusinghs) [inline packages/mongodb-dialects/spring-criteria/src/main/kotlin/com/mongodb/jbplugin/dialects/springcriteria/SpringCriteriaContextExtractor.kt:20]: Could it be that there would be both properties and yaml file? If so how should we define the priority?

File: packages/mongodb-dialects/java-driver/src/test/kotlin/com/mongodb/jbplugin/dialects/javadriver/glossary/NamespaceExtractorTest.kt [hunk 2]
```diff
@@ -438,4 +438,67 @@ public class JavaDriverRepository extends BaseRepository {
         assertEquals("myDatabase", namespace.database)
         assertEquals("myCollection", namespace.collection)
     }
+
+    @ParsingTest(
+        "Repository.java",
+        """
+import com.mongodb.client.FindIterable;
```
Comment (himanshusinghs) [inline packages/mongodb-dialects/java-driver/src/test/kotlin/com/mongodb/jbplugin/dialects/javadriver/glossary/NamespaceExtractorTest.kt:445]: This test looks not relevant to the changes here, or is it somehow?

**Review by himanshusinghs — APPROVED** [review]

File: packages/mongodb-dialects/java-driver/src/test/kotlin/com/mongodb/jbplugin/dialects/javadriver/glossary/NamespaceExtractorTest.kt (hunk 2 above)
Comment (kmruiz) [reply packages/mongodb-dialects/java-driver/src/test/kotlin/com/mongodb/jbplugin/dialects/javadriver/glossary/NamespaceExtractorTest.kt:445]: Ah no, it's not relevant, forgot to remove it after a PoC. Thanks for realising!

File: packages/mongodb-dialects/spring-criteria/src/main/kotlin/com/mongodb/jbplugin/dialects/springcriteria/SpringCriteriaContextExtractor.kt (hunk 1 above)
Comment (kmruiz) [reply packages/mongodb-dialects/spring-criteria/src/main/kotlin/com/mongodb/jbplugin/dialects/springcriteria/SpringCriteriaContextExtractor.kt:20]: Spring only allows one and it priorities the property file so it should be safe this way.

**Pull request merged**